frame/dynamic-fee/src/lib.rs

```diff
@@ -19,103 +19,133 @@
+	pub trait Config: frame_system::Config {
+		/// Bound divisor for min gas price.
+		type MinGasPriceBoundDivisor: Get<U256>;
 	}
-}
 
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		fn on_initialize(_block_number: T::BlockNumber) -> Weight {
-			TargetMinGasPrice::kill();
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_n: BlockNumberFor<T>) -> Weight {
+			TargetMinGasPrice::<T>::kill();
 
 			T::DbWeight::get().writes(1)
 		}
 
-		fn on_finalize(_n: T::BlockNumber) {
-			if let Some(target) = TargetMinGasPrice::take() {
-				let bound = MinGasPrice::get() / T::MinGasPriceBoundDivisor::get() + U256::one();
+		fn on_finalize(_n: BlockNumberFor<T>) {
+			if let Some(target) = TargetMinGasPrice::<T>::take() {
+				let bound = MinGasPrice::<T>::get() / T::MinGasPriceBoundDivisor::get() + U256::one();
 
-				let upper_limit = MinGasPrice::get().saturating_add(bound);
-				let lower_limit = MinGasPrice::get().saturating_sub(bound);
+				let upper_limit = MinGasPrice::<T>::get().saturating_add(bound);
+				let lower_limit = MinGasPrice::<T>::get().saturating_sub(bound);
 
-				MinGasPrice::set(min(upper_limit, max(lower_limit, target)));
+				MinGasPrice::<T>::set(min(upper_limit, max(lower_limit, target)));
 			}
 		}
+	}
 
-		#[weight = (T::DbWeight::get().writes(1), DispatchClass::Mandatory)]
-		pub fn note_min_gas_price_target(
-			origin,
-			target: U256,
-		) {
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight((T::DbWeight::get().writes(1), DispatchClass::Mandatory))]
+		pub fn note_min_gas_price_target(origin: OriginFor<T>, target: U256) -> DispatchResult {
 			ensure_none(origin)?;
-			assert!(TargetMinGasPrice::get().is_none(), "TargetMinGasPrice must be updated only once in the block");
+			assert!(
+				TargetMinGasPrice::<T>::get().is_none(),
+				"TargetMinGasPrice must be updated only once in the block",
+			);
 
-			TargetMinGasPrice::set(Some(target));
+			TargetMinGasPrice::<T>::set(Some(target));
+			Ok(())
 		}
 	}
-}
 
-impl<T: Config> pallet_evm::FeeCalculator for Module<T> {
-	fn min_gas_price() -> U256 {
-		MinGasPrice::get()
+	#[pallet::genesis_config]
+	pub struct GenesisConfig {
+		pub min_gas_price: U256,
 	}
-}
 
-impl<T: Config> ProvideInherent for Module<T> {
-	type Call = Call<T>;
-	type Error = InherentError;
-	const INHERENT_IDENTIFIER: InherentIdentifier = INHERENT_IDENTIFIER;
-
-	fn create_inherent(data: &InherentData) -> Option<Self::Call> {
-		let target = data.get_data::<InherentType>(&INHERENT_IDENTIFIER).ok()??;
+	#[cfg(feature = "std")]
+	impl Default for GenesisConfig {
+		fn default() -> Self {
+			Self {
+				min_gas_price: Default::default(),
+			}
+		}
+	}
 
-		Some(Call::note_min_gas_price_target(target))
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			MinGasPrice::<T>::put(self.min_gas_price);
+			MinGasPrice::<T>::set(U256::from(1));
```

Comment:
These two lines are weird, but I copied them from the generated pallet migration template. What should really be the correct behaviour here? I'm guessing it's `MinGasPrice::<T>::put(self.min_gas_price)`...

Reply:
Yeah please remove this.